goal_src/engine/debug/menu.gc

```diff
@@ -5,3 +5,1968 @@
+     arg3
+     arg4
+     )
+    )
+   ((= (-> arg0 type) debug-menu-item-function)
+    (debug-menu-item-function-render
+     (the-as debug-menu-item-function arg0)
+     arg1
+     arg2
+     arg3
+     arg4
+     )
+    )
+   ((= (-> arg0 type) debug-menu-item-flag)
+    (debug-menu-item-flag-render
+     (the-as debug-menu-item-flag arg0)
+     arg1
+     arg2
+     arg3
+     arg4
+     )
+    )
+   ((= (-> arg0 type) debug-menu-item-var)
+    (debug-menu-item-var-render
+     (the-as debug-menu-item-var arg0)
+     arg1
+     arg2
+     arg3
+     arg4
+     )
+    )
+   (else
+    (format 0 "ERROR: Found unknown item type!~%")
+    )
+   )
+  arg0
+  )
+
+;; definition for function debug-menu-render
+;; Used lq/sq
+(defun
+  debug-menu-render
+  ((arg0 debug-menu) (arg1 int) (arg2 int) (arg3 int) (arg4 int))
+  (local-vars (sv-16 dma-buffer) (sv-32 pointer))
+  (let ((v1-0 0))
+   (let* ((a0-1 (-> arg0 items))
+          (a1-1 (car a0-1))
+          )
+    (while (not (null? a0-1))
+     (if (= a1-1 arg3)
+      (begin
+       (goto cfg-7)
+       )
+      )
+     (+! v1-0 1)
+     (set! a0-1 (cdr a0-1))
+     (set! a1-1 (car a0-1))
+     )
+    )
+   (label cfg-7)
+   (if (< 16 v1-0)
+    (set! arg2 (- arg2 (* (+ v1-0 -16) 8)))
+    )
+   )
+  (let* ((s0-0 (-> *display* frames (-> *display* on-screen) frame debug-buf))
+         (s1-0 (-> s0-0 base))
+         )
+   (draw-sprite2d-xy
+    s0-0
+    arg1
+    arg2
+    (-> arg0 pix-width)
+    (-> arg0 pix-height)
+    (new 'static 'rgba :a #x40)
+    )
+   (let ((a3-2 (-> s0-0 base)))
+    (let ((v1-7 (the-as object (-> s0-0 base))))
+     (set!
+      (-> (the-as dma-packet v1-7) dma)
+      (new 'static 'dma-tag :id (dma-tag-id next))
+      )
+     (set! (-> (the-as dma-packet v1-7) vif0) (new 'static 'vif-tag))
+     (set! (-> (the-as dma-packet v1-7) vif1) (new 'static 'vif-tag))
+     (set! (-> s0-0 base) (&+ (the-as pointer v1-7) 16))
+     )
+    (dma-bucket-insert-tag
+     (-> *display* frames (-> *display* on-screen) frame bucket-group)
+     68
+     s1-0
+     (the-as (pointer dma-tag) a3-2)
+     )
+    )
+   )
+  (let* ((s3-1 (+ arg1 3))
+         (s2-1 (+ arg2 3))
+         (s1-1 (-> arg0 items))
+         (s0-1 (car s1-1))
+         )
+   (while (not (null? s1-1))
+    (when (= s0-1 arg3)
+     (set! (-> arg0 context font color) (if (nonzero? arg4)
+                                         13
+                                         12
+                                         )
+      )
+     (let ((v1-16 (-> arg0 context font))
+           (a1-5 s3-1)
+           (a0-18 s2-1)
+           )
+      (set! (-> v1-16 origin x) (the float a1-5))
+      (set! (-> v1-16 origin y) (the float a0-18))
+      )
+     (set! sv-16 (-> *display* frames (-> *display* on-screen) frame debug-buf))
+     (set! sv-32 (-> sv-16 base))
+     (let ((t9-2 draw-string)
+           (a0-22 ">")
+           (a1-6 sv-16)
+           )
+      (-> arg0 context font)
+      (t9-2 a0-22 a1-6)
```

Comment:
function missing arguments here

Reply:
hmm this looks like an original bug.  Looks like they meant to use `draw-string-adv` which takes a `font-context` as a third argument.  `draw-string` looks like it only takes 2 arguments (as seen in debug.gc)